lib/galaxy/config/__init__.py

```diff
@@ -435,22 +422,19 @@ def _process_config(self, kwargs):
         self.enable_mulled_containers = string_as_bool(kwargs.get('enable_mulled_containers', 'True'))
         containers_resolvers_config_file = kwargs.get('containers_resolvers_config_file', None)
         if containers_resolvers_config_file:
-            containers_resolvers_config_file = resolve_path(containers_resolvers_config_file, self.root)
+            containers_resolvers_config_file = os.path.join(self.root, containers_resolvers_config_file)
         self.containers_resolvers_config_file = containers_resolvers_config_file
 
-        involucro_path = kwargs.get('involucro_path', None)
-        if involucro_path is None:
-            target_dir = kwargs.get("tool_dependency_dir", "dependencies")
-            if target_dir == "none":
-                target_dir = "dependencies"
-            target_dir = resolve_path(target_dir, self.data_dir)
-            involucro_path = os.path.join(target_dir, "involucro")
-        self.involucro_path = resolve_path(involucro_path, self.root)
-        mulled_channels = kwargs.get('mulled_channels')
-        if mulled_channels:
-            self.mulled_channels = [c.strip() for c in mulled_channels.split(',')]
-        else:
-            self.mulled_channels = DEFAULT_CHANNELS
+        # tool_dependency_dir can be "none" (in old configs). If so, set it to schema default
+        if self.tool_dependency_dir and self.tool_dependency_dir.lower() == 'none':
+            self.tool_dependency_dir = self.appschema['tool_dependency_dir'].get('default')
```

Comment:
According to `tool_dependency_dir` documentation in `lib/galaxy/webapps/galaxy/config_schema.yml` , it seems `None` and `none` can be used to disable tool dependency resolution, see also https://github.com/galaxyproject/galaxy/blob/dev/lib/galaxy/tool_util/deps/__init__.py#L51-L52

Ping @jmchilton 

```suggestion
            self.tool_dependency_dir = None
```

Reply:
Yes, this makes sense. Not sure what I was thinking.

Reply:
It looks like this change broke some integration tests: in the old version tool_dependency_dir was never None, but now it can be - and so `os.path.join` got a NoneType as a result. I'll fix this today.

Reply:
Also, after this line, should `self.tool_dependency_dir` be resolved w.r.t. to `self.data_dir` if it is not None?
At the moment it's only used for the involucro subdir, but maybe better to have it fixed if it is more used in the future?

Reply:
Very possibly. But I'd rather not do it here. I don't understand its use well enough, and it is accessed + used for path resolving in multiple files. It was not resolved here prior to these changes, so I'd rather leave it as is for now: I am simply not sure I wouldn't be breaking something. Also, I'd rather not introduce new changes in the same PR, if possible.